rosbag2_cpp/src/rosbag2_cpp/reindexer.cpp

```diff
@@ -188,6 +190,17 @@ void Reindexer::aggregate_metadata(
     auto temp_metadata = bag_reader->get_metadata();
     metadata_.storage_identifier = temp_metadata.storage_identifier;
 
+    // try to find the last log for the most complete custom data section
+    if (latest_log_start < temp_metadata.starting_time ) {
+      latest_log_start = temp_metadata.starting_time;
+      if (temp_metadata.custom_data.size() > 0) {
+        metadata_.custom_data = temp_metadata.custom_data;
+      }
+      if (temp_metadata.ros_distro.size() > 0) {
+        metadata_.ros_distro = temp_metadata.ros_distro;
+      }
+    }
```

Comment:
Please use the `.empty()` property of the containers instead of the `size()` for checking if it is empty or not.
Also, please address `uncrustify` warning.

Reply:
Switched to use empty and fixed the `uncrustify` warning as requested.